pytorch_lightning/accelerators/accelerator.py

```diff
@@ -24,6 +24,11 @@
 from pytorch_lightning.utilities.distributed import all_gather_ddp_if_available
 from pytorch_lightning.utilities.enums import AMPType, LightningEnum
 
+if TYPE_CHECKING:
+    from torch.cuda.amp import GradScaler
+
+    from pytorch_lightning.trainer.trainer import Trainer
```

Comment:
do you need this even you annotate it with a string?

Reply:
Yes, because if we annotate with the plain class it still has to be imported globally, since else there are undefined variables during Runtime. MyPy then reads those strings as class variables